tests/clients/test_fenda_client.py

```diff
@@ -42,6 +48,91 @@ def test_getting_parameters(get_fenda_client: FendaClient) -> None:  # noqa
         assert (params_global[i] == old_global_module_params[i]).all()
 
 
+@pytest.mark.parametrize("local_module,global_module,head_module", [(FeatureCnn(), FeatureCnn(), FendaHeadCnn())])
+def test_setting_global_model(get_fenda_client: FendaClient) -> None:  # noqa
+    torch.manual_seed(42)
+    fenda_client = get_fenda_client
+
+    assert fenda_client.aggregated_global_module is None
+    assert isinstance(fenda_client.model, FendaModel)
+
+    global_params = [
+        copy.deepcopy(val.cpu().numpy()) for _, val in fenda_client.model.global_module.state_dict().items()
+    ]
+    fenda_client.update_before_train(0)
+
+    assert fenda_client.aggregated_global_module is not None
+
+    aggregate_params = [
+        copy.deepcopy(val.cpu().numpy()) for _, val in fenda_client.aggregated_global_module.state_dict().items()
+    ]
+    # Make sure the fenda aggregated module parameters are equal to the global module parameters
+    for i in range(len(aggregate_params)):
+        assert (aggregate_params[i] == global_params[i]).all()
+
+    # Make sure the aggregated module is not set to train
+    assert fenda_client.aggregated_global_module.training is False
+    for param in fenda_client.aggregated_global_module.parameters():
+        assert param.requires_grad is False
+
+    # Make sure the original model is still set to train
+    assert fenda_client.model.training is True
+    for param in fenda_client.model.parameters():
+        assert param.requires_grad is True
```

Comment:
You should unset the seed at the end or we risk having the seed set for all the tests that run after this one. See https://github.com/VectorInstitute/FL4Health/blob/aeb4d125a0c06bbddb5ce3d65c07e46a90fab444/tests/clients/test_moon_client.py#L35